src/util.cxx

```diff
@@ -9,14 +9,139 @@
 
 #include "util.h"
 
+#include <vector>
+#include <cctype>
+#include <cmath>
+
 NAMESPACE__THITTAM_UTIL__START
 
+
+std::string getString(const std::string &s, unsigned &pos) {
+  std::string returnValue = "";
+  while (isalpha(s[pos]))
+    {
+      returnValue += tolower(s[pos]);
+      pos++;
+    }
+  return returnValue;
+}
+
+
+double getNumber(const std::string &s, unsigned &pos) {
+  double x = 0.0;
+  bool sign = 0;
+
+  if (s[pos] == '-')
+    sign = true, pos++;
+  else if (s[pos] == '+')
+    sign = false, pos++;
+
+  while (isdigit(s[pos]))
+    {
+      // Convert the integers first
+      x = x * 10 + s[pos] - '0';
+      pos++;
+    }
+  if (s[pos] == '.')
+    {
+      // Then the fraction
+      pos++;
+      int negPow = 0;
+      while (isdigit(s[pos]))
+        {
+          if (negPow > -6)
+            x = x + (s[pos] - '0') * pow(10, --negPow);
+          pos++;
+        }
+    }
+  x = round(x * 1e4) / 1e4;   // 4 digits of significance after decimal point
+  return sign ? -x : x;
+}
+
+
+int parse(const std::string &s) {
+  if (s == "y" || s == "yr" || s == "yrs" || s == "year" || s == "years")
+    return 'y';  // <-------------year
+
+  if (s == "m"    || s == "month" || s == "months" || s == "mth" ||
+      s == "mths" || s == "mnth"  || s == "mnths")
+    return 'M';  // <-------------month
+
+  if (s == "w" || s == "wk" || s == "wks" || s == "wek" || s == "weks" ||
+      s == "week" || s == "weeks")
+    return 'w';  // <-------------week
+
+  if (s == "d"   || s == "da" || s == "das" || s == "dy" || s == "dys" ||
+      s == "day" || s == "days")
+    return 'd';  // <-------------day
+
+  if (s == "h" || s == "hr" || s == "hrs" || s == "hour" || s == "hours")
+    return 'h';  // <-------------hour
+
+  if (s == "mn" || s == "mns" || s == "min" || s == "mins" ||
+      s == "minute" || s == "minutes")
+    return 'm';  // <-------------minute
+
+  if (s == "s" || s == "sec" || s == "secs" || s == "seconds" || s == "second")
+    return 's';  // <-------------second
+
+  return 0;
+}
+
+
+// This function returns the minutes given the effort like "1d 3.5h". It throws
+// error only when days_per_week or hours_per_day is an invalid number.
 int effort_str_to_minutes (
-  const std::string & effort, int working_days_per_week,
-  int working_hours_per_day)
+  const std::string & effort, int days_per_week,
+  int hours_per_day)
 {
-  // todo: issue #8
-  return 2400;
+
+  double minutes = 0;
+
+  double currentNumber = 0;
+  std::string currentString = "";
+
+  for (unsigned i = 0; i < effort.length(); ++i)
+    {
+
+      while (isspace(effort[i])) i++;  // Skip space
```

Comment:
You should NEVER do this. This is very tricky for readability. Its much better to just 'continue'. That way the control of the iteration is in one place, i..e the 'for'.